client/rest-high-level/src/main/java/org/elasticsearch/client/dataframe/transforms/pivot/DateHistogramGroupSource.java

```diff
@@ -168,4 +168,55 @@ public boolean equals(Object other) {
     public int hashCode() {
         return Objects.hash(field, interval, dateHistogramInterval, timeZone, format);
     }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+
+        private String field;
+        private long interval = 0;
+        private DateHistogramInterval dateHistogramInterval;
+        private String format;
+        private ZoneId timeZone;
+
+        public Builder setField(String field) {
+            this.field = field;
+            return this;
+        }
+
+        public Builder setInterval(long interval) {
+            this.interval = interval;
+            return this;
+        }
+
+        public Builder setDateHistgramInterval(DateHistogramInterval interval) {
+            this.dateHistogramInterval = interval;
+            return this;
+        }
+
+        public Builder setFormat(String format) {
+            this.format = format;
+            return this;
+        }
+
+        public Builder setTimeZone(ZoneId timeZone) {
+            this.timeZone = timeZone;
+            return this;
+        }
+
+        public DateHistogramGroupSource build() {
+            DateHistogramGroupSource groupSource = new DateHistogramGroupSource(field);
```

Comment:
Introducing Builder is a great opportunity to make the data class itself (here, DateHistogramGroupSource) immutable. Do you plan on making it so or you find such an approach too restrictive?

Reply:
We totally should do this. I will see if I can make them `final`.